contracts/extensions/VotingBase.sol

```diff
@@ -0,0 +1,1108 @@
+  function deprecate(bool _deprecated) public override auth {
+    deprecated = _deprecated;
+  }
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  // Data structures
+  enum MotionState { Null, Staking, Submit, Reveal, Closed, Finalizable, Finalized, Failed }
+
+  struct Motion {
+    uint64[3] events; // For recording motion lifecycle timestamps (STAKE, SUBMIT, REVEAL)
+    bytes32 rootHash;
+    uint256 domainId;
+    uint256 skillId;
+    uint256 paidVoterComp;
+    uint256[2] pastVoterComp; // [nay, yay]
+    uint256[2] stakes; // [nay, yay]
+    uint256[2][] votes; // [nay, yay]
+    uint256[] totalVotes;
+    uint256[] maxVotes;
+    bool escalated;
+    bool finalized;
+    address altTarget;
+    bytes action;
+  }
+
+  // Storage
+  uint256 motionCount;
+  mapping (uint256 => Motion) motions;
+  mapping (uint256 => mapping (address => mapping (uint256 => uint256))) stakes;
+  mapping (uint256 => mapping (address => bytes32)) voteSecrets;
+
+  mapping (bytes32 => uint256) expenditurePastVotes; // expenditure slot signature => voting power
+  mapping (bytes32 => uint256) expenditureMotionCounts; // expenditure struct signature => count
+
+  // Virtual functions
+
+  function getInfluence(uint256 _motionId, address _user) public view virtual returns (uint256[] memory);
+  function postReveal(uint256 _motionId, address _user) internal virtual;
+  function postClaim(uint256 _motionId, address _user) internal virtual;
+
+  // Public functions (interface)
+
+  /// @notice Reveal a vote secret for a motion
+  /// @param _motionId The id of the motion
+  /// @param _salt The salt used to hash the vote
+  /// @param _vote The side being supported (0 = NAY, 1 = YAY)
+  function revealVote(uint256 _motionId, bytes32 _salt, uint256 _vote) public {
+    Motion storage motion = motions[_motionId];
+
+    require(getMotionState(_motionId) == MotionState.Reveal, "voting-base-motion-not-reveal");
+    require(_vote <= 1, "voting-base-bad-vote");
+
+    uint256[] memory influence = getInfluence(_motionId, msg.sender);
+    for (uint256 i; i < motion.votes.length; i++) {
+      motion.votes[i][_vote] = add(motion.votes[i][_vote], influence[i]);
+    }
+
+    bytes32 voteSecret = voteSecrets[_motionId][msg.sender];
+    require(voteSecret == getVoteSecret(_salt, _vote), "voting-base-secret-no-match");
+    delete voteSecrets[_motionId][msg.sender];
+
+    uint256 voterReward = getVoterReward(_motionId, msg.sender);
+    motion.paidVoterComp = add(motion.paidVoterComp, voterReward);
+
+    emit MotionVoteRevealed(_motionId, msg.sender, _vote);
+
+    // See if reputation revealed matches reputation submitted
+    bool fullyRevealed = true;
+
+    for (uint256 j; j < motion.totalVotes.length; j++) {
+      fullyRevealed = fullyRevealed &&
+        add(motion.votes[j][NAY], motion.votes[j][YAY]) == motion.totalVotes[j];
```

Comment:
Again, once this is false, we don't have to go through the rest of the loop, same for other loops in this function.

Reply:
There are arguments against this optimisation I suppose, but it's a discussion we should have regardless.

Reply:
I'm pro, I don't see any reasons not to break out of these loops early if possible.